src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/EventPipeProvider.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public class EventPipeProvider
+    {
+        public EventPipeProvider(string name, EventLevel eventLevel, long keywords = 0, IDictionary<string, string> arguments = null)
+        {
+            Name = name;
+            EventLevel = eventLevel;
+            Keywords = keywords;
+            Arguments = arguments;
+        }
+
+        public long Keywords { get; }
+
+        public EventLevel EventLevel { get; }
+
+        public string Name { get; }
+
+        public IDictionary<string, string> Arguments { get; }
+
+        public override string ToString()
+        {
+            return $"{Name}:0x{Keywords:X16}:{(uint)EventLevel}{(Arguments == null ? "" : $":{GetArgumentString()}")}";
+        }
+        
+        public override bool Equals(object obj)
```

Comment:
nit: When overriding the `Equals` method follow this pattern:
```diff
- public override bool Equals(object obj)
- {
-     if (obj == null || GetType() != obj.GetType())
-     {
-         return false;
-     }
- 
-     return this == (EventPipeProvider)obj;
- }
+ public override bool Equals(object obj)
+ {
+    return Equals(obj as EventPipeProvider);
+ }
+ 
+ public bool Equals(EventPipeProvider other)
+ {
+     if ((object)other == null)
+         return false;
+     return this == other;
+ }
```

Reply:
Thanks! This + the comment below together would cause StackOverflowException though :p I'll modify it accordingly

Reply:
I'm not sure that you want to do this: `this == other`

Reply:
Yeah. The `==` operator does that (or used to). Essentially these eventually get dumped out to a string and that's what gets used so I check using the `ToString()` method.